.github/workflows/ci.yaml

```diff
@@ -41,12 +45,35 @@ jobs:
         python-version: ["3.12", "3.13"]
 
     steps:
-      - uses: actions/checkout@v4
+      - name: Check out repo
+        uses: actions/checkout@v4
+
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+
       - name: Install uv
         uses: astral-sh/setup-uv@v5
+
       - name: Run unit tests
         run: make unit
+
+  integration:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+
+      - name: Install concierge
+        run: sudo snap install --classic concierge
+
+      - name: Prepare Juju
+        run: sudo concierge prepare --verbose --juju-channel=3/stable --charmcraft-channel=3.x/stable -p microk8s
+
+      - name: Run integration tests
+        run: make integration
```

Comment:
Off-topic: I've been using shorthand like this:

```suggestion
    steps:
      - uses: actions/checkout@v4
      - uses: astral-sh/setup-uv@v5
      - run: sudo snap install --classic concierge
      - run: sudo concierge prepare --verbose --juju-channel=3/stable --charmcraft-channel=3.x/stable -p microk8s
      - run: make integration
```

Which is both concise and expressive:

![Screenshot 2025-03-10 at 13 10 12](https://github.com/user-attachments/assets/cc7675a9-0c10-4008-95e6-55a4f38852fc)

☝🏻  https://github.com/dimaqq/otlp-json/actions/runs/13646276620/job/38145683050

Reply:
Interesting idea. I've looked at it both ways locally, and initially I quite liked your terse approach, but then it kinda broke down for `run: sudo concierge prepare --verbose --juju-channel=3/stable --charmcraft-channel=3.x/stable -p microk8s`. I think the nice names are okay, and I don't mind the increased verbosity here, so I'm going to leave them.